src/components/Filter/types.ts

```diff
@@ -0,0 +1,71 @@
+import { AcceptedColorComponentTypes, ColorShapeFromComponent } from '../../utils/themeFunctions';
+import * as React from 'react';
+import { Theme } from '../../theme';
+
+export type FilterOption = {
+  value: string | number;
+  label: string;
+};
+
+export type StyleType = 'filled' | 'outlined' | 'elevated' | 'transparent';
+
+export type Props = {
+  /** the color of the button when the menu is opened based on our colors eg. red-400 */
+  color?: string;
+  /** The type of the button - defaults to "primary" */
+  buttonType?: AcceptedColorComponentTypes;
+  /** Items that are being declared as menu options */
+  items?: FilterOption[];
+  /** The item selected on the menu */
+  selectedItem: FilterOption;
+  /** The default selected item */
+  defaultValue: FilterOption;
+  /** A callback that is being triggered when an items has been clicked */
+  onSelect: (option: FilterOption) => void;
+  /** The text of the button to show - defaults to "More" */
+  label: React.ReactNode;
+  /** Defines if the button is in disabled state */
+  disabled?: boolean;
+  /** Defines the style type of the button */
+  styleType: StyleType;
+};
+
+export type TestProps = {
+  dataTestId?: string;
+};
+
+
+export type BackgroundColorProps = {
```

Comment:
as you use often the following

```
open: boolean;
  theme: Theme;
  hasSelectedValue: boolean;
  calculatedColor: ColorShapeFromComponent;
  activeCalculatedColor: ColorShapeFromComponent;
```
you can make a base type and use this as extension